odl/operator/fn_ops.py

```diff
@@ -0,0 +1,406 @@
+## Copyright 2014-2016 The ODL development group
+##
+## This file is part of ODL.
+##
+## ODL is free software: you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## ODL is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+#
+#"""Default operators defined on any (reasonable) space."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+from odl.operator.operator import Operator
+from odl.space import fn
+from odl.set import LinearSpace, Field
+#from odl.util import writable_array
+
+import numpy as np
+
+__all__ = ('SamplingOperator', 'ZerofillingOperator', 
+           'VectoriseOperator', 'EmbeddingOperator')
+
+class SamplingOperator(Operator):
+
+    """Operator that samples coefficients.
+
+        ``SamplingOperator(f)(i) == cell_volume * f(i) \approx \int_cell f(x) dx``
```

Comment:
I agree about the syntax and I have changed that. 

About the cell_volume I disagree as we are trying to approximate \int_{\text{cell}_{indices_i}} f(x) dx ??? But then this should depend on how the interpolation is done as well which currently it does not. The current version is the right formula if we assume that f is constant over the cell.